### PR TITLE
chore(build): Provide multiple build jobs.  Closes #150

### LINKED ISSUE #150: Split CI build in separate jobs for clearer overview and restart
**Is your feature request related to a problem? Please describe.**
Currently the CI build is one big job which doesn't provide much clarity in the PR overview and does not provide more gradual control if a job needs to be restarted.

**Describe the solution you'd like**
Split the CI build in separate jobs (like in the Arcus repo's).

### PR DESCRIPTION
**Pull request by SamVanhoutte**
Add multiple stages & jobs:

- Prerequisites
- UnitTests
- PackageAndRelease

### CODE REVIEW

**Comment by SamVanhoutte** [comment]
Build seems to fail, because of dependencies installed in first stage, not being found in the second stage.
Is this because they could be seperate machines? 

image

**Comment by stijnmoreels** [comment]
> Build seems to fail, because of dependencies installed in first stage, not being found in the second stage. Is this because they could be seperate machines?
> 
> image

Yeah, but if you can install them locally into a build package you can pass along them to the next stage. Our other Arcus repo's all have this approach. We'll see.

**Review by stijnmoreels — COMMENTED** [review]
Great to have some more separation and consistency with the Arcus repo's. I'm wondering if the other Python ML package uses the same (almost the same) setup for building; and if so, we could move it to our general Azure DevOps YAML templates repo instead of manually having all the script commands here.

Also, we should update the release pipeline too, right?

**Comment by SamVanhoutte** [comment]
@stijnmoreels release build is updated.  
Issues https://github.com/arcus-azure/azure-devops-templates/issues/50 and https://github.com/arcus-azure/azure-devops-templates/issues/51 created for the template generation

**Comment by stijnmoreels** [comment]
> @stijnmoreels release build is updated. Issues [arcus-azure/azure-devops-templates#50](https://github.com/arcus-azure/azure-devops-templates/issues/50) and [arcus-azure/azure-devops-templates#51](https://github.com/arcus-azure/azure-devops-templates/issues/51) created for the template generation

Great! 👍

**Comment by stijnmoreels** [comment]
We can always look further into why the build and tests jobs needs to be in the same stage, but for a first run, I think this is good. @ALaks96 , what do you think?

File: build/ci-build.yml [hunk 1]
```diff
@@ -27,73 +29,86 @@ resources:
...
+      - job: DefineVersion
+        displayName: 'Increase and define version number'
+        steps:
+          - template: 'pypi/determine-pr-version.yml@templates'
+            parameters:
+              manualTriggerVersion: preview
+
+          - task: UsePythonVersion@0
+            inputs:
+              versionSpec: '3.7'
```
Comment (stijnmoreels) [inline build/ci-build.yml:45]: Something we can place in a YAML variable template later.

**Review by stijnmoreels — APPROVED** [review]
Great first improvement on this 💯

File: build/ci-build.yml (hunk 1 above)
Comment (SamVanhoutte) [reply build/ci-build.yml:45]: yes, agreed.

**Pull request merged**